### ISSUE TITLE
deps parameter not working in maven_aar rule

### ISSUE THREAD

**Issue by izzyleung** [bazelbuild/bazel#4294]
### Description of the problem / feature request / question:

While trying to build an Android app using bazel, I have to use the `design` library which depends on `appcompat`.
If trying to build the app using the `bazel build //:android` command, while not including the `deps` parameter for `maven_aar` in https://gist.github.com/izzyleung/26ea8e36a6dc4ea7e55cff704c61fecf#file-workspace-L20, bazel will yield the following error message:

```
ERROR: /..../maven_aar_test/BUILD:1:1: Merging Android resources for //:activities failed (Exit 1)
Dec 13, 2017 4:22:46 PM com.google.devtools.build.android.AndroidResourceMergingAction main
SEVERE: Error during merging resources
com.google.devtools.build.android.AndroidResourceMerger$MergingException: java.io.IOException: com.google.devtools.build.android.AndroidFrameworkAttrIdProvider$AttrLookupException: App attribute not found: elevation
```

That is due to the reason that bazel was not able to find the related resources file that is located in the `appcompat` library.

__Related issues:__ https://github.com/bazelbuild/bazel/issues/3980

@aj-michael updated the `maven_aar` rules, which is reflected here at: https://github.com/bazelbuild/bazel/blob/c6557ea56476fd56ee94f2476ac01849841bcaf0/tools/build_defs/repo/maven_rules.bzl#L167-L169

After the latest release, the `maven_aar` rule should be able to accept the parameter `deps`.

However, after including the `deps` parameter in `maven_aar`, bazel will yield the following error message:

```
ERROR: /......./maven_aar_test/WORKSPACE:15:1: //external:design: no such attribute 'deps' in 'maven_aar' rule
ERROR: Skipping '//:maven_aar_deps_test': error loading package 'external': Package 'external' contains errors
WARNING: Target pattern parsing failed.
ERROR: error loading package 'external': Package 'external' contains errors
```

### If possible, provide a minimal example to reproduce the problem:

https://gist.github.com/izzyleung/26ea8e36a6dc4ea7e55cff704c61fecf

### Environment info
* Operating System:

macOS Sierra 10.12.6

* Bazel version (output of `bazel info release`):

`release 0.8.1-homebrew`

**Comment by izzyleung** [comment]
Sorry, I have not yet checkout the release log.

The change will take effect in release 0.9.0.

If anyone is bother by this problem, try using the 0.9.0 release candidate.

**Comment by aj-michael** [comment]
Sorry for the confusion here. 0.8.1 was a patch release to fix a bug in 0.8.0 so it did not have new features. Glad to hear that 0.9.0 is working for you.

**Comment by izzyleung** [comment]
@aj-michael 

To resolve this issue, I just copy and paste the latest version of `maven_rules.bzl` to my local machine, and now, bazel will report a new error message.

I wonder if you can take a look at the updated gist at: https://gist.github.com/izzyleung/26ea8e36a6dc4ea7e55cff704c61fecf

The thing to take notice is: https://gist.github.com/izzyleung/26ea8e36a6dc4ea7e55cff704c61fecf#file-errormessage-md
